flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/TableSchema.java

```diff
@@ -421,9 +421,9 @@ private static void validateColumnsAndWatermarkSpecs(
 			LogicalType watermarkOutputType = watermark.getWatermarkExprOutputType().getLogicalType();
 			if (watermarkOutputType.getTypeRoot() != TIMESTAMP_WITHOUT_TIME_ZONE) {
 				throw new ValidationException(String.format(
-					"Watermark strategy '%s' must be of type TIMESTAMP but is of type '%s'.",
```

Comment:
Why remove `''`?

Reply:
I added a test for empty char. The exception message will be `Watermark strategy '''' must be`...